### PR TITLE
Dark Theme #55

### PR DESCRIPTION
**Pull request by vitalragaz**
**Description:**
This PR implements the requested dark mode support #55 for CapRover. The preference is stored within the local storage, if no exists a fallback to the preference scheme of the system is used. The switch for the mode is located in the navigation bar. The approach used for the implementation is based on the [antd-theme-webpack-plugin](https://github.com/mzohaibqc/antd-theme-webpack-plugin) as it is the most common way for dynamic themes with antd. In order to keep changes small as also to prevent a redux overwhelming implementation, .css files were transformed into .less to use the colour vars from antd for project-specific styling.

**Preview:**
CapRover _ Server Dashboard

### CODE REVIEW

**Comment by githubsaturn** [comment]
Thanks! Looks great! One question:
`npm start` takes a very long time. What change causes this huge uptick? Can we optimize it?

**Comment by vitalragaz** [comment]
I have already observed the same thing. I have analysed how much the difference is to the current master. However, it turns out that the 0.5s is justified, it looks like it comes from an earlier commit.

Startup time of "Starting the development server..." measured by gnomon.

| Master branch | Dark_Mode branch |
|--------|-----------|
| 14.7s   |   14.9s        |
|  15.1s      |   15.4s        |
|    14.6s    |      15.5s     |
|   Avg:  **14.8s**    |  Avg:    **15.3s**    |

**Comment by githubsaturn** [comment]
Ah yes, you're correct. 

Fantastic work though! This looks great!

Before we merge this, can you please a few lines as the "dev guide" to readme file for future developers. Looking at the variables, it's not immediately obvious what the variables are, `@layout-body-background` `@body-background` `@skeleton-color`  and etc....

Something super basic would be great, something like:
```
To ensure your elements work with both dark and light modes:
- Use "@layout-body-background" for all page backgrounds like xxx and xxx
- Use "@skeleton-color" for all logs backgrounds like app logs etc
```

**Comment by githubsaturn** [comment]
Thanks @vitalragaz ! Fantastic work!

File: src/App.less [hunk 1]
```diff
@@ -5,6 +5,10 @@
     overflow-x: hidden;
 }
 
+body when (@theme = 'dark') {
```
Comment (githubsaturn) [inline src/App.less:8]: Ah I noticed this change after I merged, is this needed or was it added by mistake?

Comment (vitalragaz) [reply src/App.less:8]: Oh no, my mistake. Sorry. This needs to be removed. If you shortly can do this directly on the master. I would be glad.

Comment (githubsaturn) [reply src/App.less:8]: Cool! Fixed on master: https://github.com/caprover/caprover-frontend/commit/42aecf94702de17e03738051b30e1b7165a171b8

Comment (vitalragaz) [reply src/App.less:8]: Cool! Thanks :)

**Comment by githubsaturn** [comment]
@vitalragaz  - I've been trying to update the build toolchain and remove `customize-cra` and `react-app-rewired` and replace them with `craco` as it seems to be the more flexible one moving forward. 

See this PR: https://github.com/caprover/caprover-frontend/pull/79

But it looks like `antd-theme-webpack-plugin` doesn't play well with it. Overrides in `./src/styles/vars.less` don't get used and everything is just using the default antd theme. If I add `themeVariables` to `modifyVars` option, the customization works, but the theme switch still doesn't work. 

When you have a chance, please have a look at the PR to see if you can spot the issue and fix it. I've been fighting with this for an entire day and I'm out of options at this point.

Essentially we want to change:
```js
module.exports = require('customize-cra')
.override(
    fixBabelImports('import', {
        libraryName: 'antd',
        libraryDirectory: 'es',
        style: true,
    }),
    addWebpackPlugin(new AntDesignThemePlugin(options)),
    addLessLoader({
        javascriptEnabled: true,
    })
) 
```
 to something like
``` js
module.exports = {
  webpack: {
    plugins: [
      new AntDesignThemePlugin(options)
    ]
  },
  plugins: [
    {
      plugin: CracoAntDPlugin,
      options: {
        lessLoaderOptions: {
          lessOptions: {
            javascriptEnabled: true,
          },
        },
        babelPluginImportOptions: {
          libraryName: 'antd',
          libraryDirectory: 'es',
          style: true
        },
      },

    },
  ],
}; 
```

**Pull request merged**